src/client/dfs/dfs.c

```diff
@@ -1209,30 +1204,35 @@ dfs_cont_create(daos_handle_t poh, uuid_t co_uuid, dfs_attr_t *attr,
 			D_GOTO(err_prop, rc);
 		}
 	}
-	if (attr && attr->da_oclass_id != OC_UNKNOWN)
+
+	/** set the oclass id from passed in attr, otherwise use default (0) */
+	if (attr)
 		dattr.da_oclass_id = attr->da_oclass_id;
 	else
-		dattr.da_oclass_id = DFS_DEFAULT_OBJ_CLASS;
+		dattr.da_oclass_id = 0;
+
+	/** check if RF factor is set on property */
+	rf_factor = daos_cont_prop2redunfac(prop);
 
 	/* select oclass and generate SB OID */
-	rc = dfs_oclass_select(poh, OC_RP_XSF, &oclass);
-	if (rc) {
-		rc = daos_der2errno(rc);
-		D_GOTO(err_prop, rc);
-	}
 	roots.cr_oids[0].lo = RESERVED_LO;
 	roots.cr_oids[0].hi = SB_HI;
-	daos_obj_set_oid(&roots.cr_oids[0], 0, oclass, 0);
-
-	/* select oclass and generate ROOT OID */
-	rc = dfs_oclass_select(poh, dattr.da_oclass_id, &oclass);
+	rc = daos_obj_generate_oid_by_rf(poh, rf_factor, &roots.cr_oids[0],
+					 0, dattr.da_oclass_id, 0, 0);
 	if (rc) {
-		rc = daos_der2errno(rc);
-		D_GOTO(err_prop, rc);
+		D_ERROR("Failed to generate DB OID "DF_RC"\n", DP_RC(rc));
+		return daos_der2errno(rc);
 	}
+
+	/* select oclass and generate ROOT OID */
 	roots.cr_oids[1].lo = RESERVED_LO;
 	roots.cr_oids[1].hi = ROOT_HI;
-	daos_obj_set_oid(&roots.cr_oids[1], 0, oclass, 0);
+	rc = daos_obj_generate_oid_by_rf(poh, rf_factor, &roots.cr_oids[1],
+					 0, dattr.da_oclass_id, 0, 0);
+	if (rc) {
+		D_ERROR("Failed to generate DB OID "DF_RC"\n", DP_RC(rc));
```

Comment:
SB too?

Reply:
this is supposed to be root.. let me fix this